product_pricelist_margin/wizards/wizard_preview_pricelist_margin.py

```diff
@@ -0,0 +1,98 @@
+# @author: Sylvain LE GAL (https://twitter.com/legalsylvain)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, fields, models
+from odoo.addons import decimal_precision as dp
+
+
+class WizardPreviewPricelistMargin(models.TransientModel):
+    _name = "wizard.preview.pricelist.margin"
+    _description = "Wizard - Preview Pricelist Margin"
+
+    template_id = fields.Many2one(
+        comodel_name="product.template", string="Product",
+        default=lambda x: x._default_template_id())
+
+    product_id = fields.Many2one(
+        comodel_name="product.product", string="Product Variant",
+        default=lambda x: x._default_product_id(),
+        domain="[('product_tmpl_id', '=', template_id)]",
+        required=True)
+
+    standard_price = fields.Float(
+        string='Cost',
+        digits=dp.get_precision('Product Price'),
+        related="product_id.standard_price",
+        readonly=True,
+        groups="base.group_user")
+
+    product_qty = fields.Float(
+        string="Quantity",
+        default=1,
+        required=True)
+
+    price_date = fields.Date(
+        string="Date", default=fields.Date.context_today,
+        required=True)
+
+    line_ids = fields.One2many(
+        comodel_name="wizard.preview.pricelist.margin.line", string="Lines",
+        inverse_name="wizard_id", compute="_compute_line_ids",
+        readonly=True)
+
+    def _default_product_id(self):
+        if self.env.context.get("active_model", False) == "product.product":
+            return self.env.context.get("active_id", False)
+        elif self.env.context.get("active_model", False) == "product.template":
+            return self.env["product.template"].browse(
+                self.env.context.get("active_id", False)
+            ).mapped("product_variant_ids")[0].id
+
+    def _default_template_id(self):
+        if self.env.context.get("active_model", False) == "product.template":
+            return self.env.context.get("active_id", False)
+
+    def _get_pricelists(self):
+        """Overload this function to filter pricelist to be displayed"""
+        return self.env["product.pricelist"].search([])
+
+    @api.depends("product_id", "product_qty", "price_date")
+    def _compute_line_ids(self):
+        pricelists = self._get_pricelists()
+        lines_vals = []
+        if not self.product_id:
+            return
+        for pricelist in pricelists:
+            product = self.product_id.with_context(
+                quantity=self.product_qty,
+                pricelist=pricelist.id,
+            )
+            to_currency = pricelist.currency_id
+            res = product.taxes_id.compute_all(
+                product.price,
+                to_currency,
+                self.product_qty,
+                product=product
+            )
+
+            price_vat_excl = res["total_excluded"] / self.product_qty
+            price_vat_incl = res["total_included"] / self.product_qty
```

Comment:
Here it should be checked that the product quantity is > 0, otherwise it triggers an error.

Reply:
Thanks ! 
Done. (i added ``if not self.product_id or not self.product_qty:`` at the beginning of the function.

Reply:
Nice, thank you! Just one question: in the margin lines displayed in the wizard, why should them be editable?

Reply:
I dont understand. There are not editable. Line_ids is set as readonly

Reply:
I mean when clicking on one line for example, the form view is opened and the user can click on Save & New, and then a blank line gets added in the form and it gets a bit weird.

![image](https://user-images.githubusercontent.com/71635103/125783187-3d586d7f-f0a4-4eed-a1dd-196f76f219b3.png)

![image](https://user-images.githubusercontent.com/71635103/125783211-553ca956-a63d-48da-b38c-8ca5a6e8bd02.png)

Reply:
That's weird and seems to be a odoo bug. As the field is readonly... do you know how to prevent it ?

Reply:
I think that it is because there is no form view for the wizard line model, so Odoo generates this one. I think that if you define a form view for the wizard lines with just the fields that are important for the user then it should be fine. Also, I think that you should do the same you did for the main wizard form view, adding explicitly the footer with the "Close" button, otherwise the other default standard buttons will appear :)

Reply:
Thanks ! 
Well, I tried some ``editable="false" edit="false" create="false" delete="false"`` without success.
Your solution works ! Could you review it ?
Well, it's a pity to have to add 15 useless xml lines just to avoid a "create and new" button on a readonly field. But well !